lib/apiconfig/apiconfig.go

```diff
@@ -91,6 +93,10 @@ func NewCalicoAPIConfig() *CalicoAPIConfig {
 		TypeMeta: metav1.TypeMeta{
 			Kind:       KindCalicoAPIConfig,
 			APIVersion: apiv3.GroupVersionCurrent,
+		},Spec: CalicoAPIConfigSpec{
```

Comment:
This is the final version. Setting `kubeconfig` is necessary imo because it can help when no configuration can be found and instead of returning an error we use Kubernetes config file for connection.

The only scenario that I could've think of for default value of `kubeconfig` as being problematic is when user sets `K8sAPIEndpoint` manually. However, since `K8sAPIEndpoint` takes precedence over `kubeconfig` it should be safe to use it like this.

> [Additionally, we probably want to change the defaults listed in the struct tags.](https://github.com/projectcalico/libcalico-go/pull/1279#pullrequestreview-467187632)

Since this is a default value I thought it is safe to set in struct tags to address @caseydavenport  request.
It can be moved to [load.go line 43 too](https://github.com/projectcalico/libcalico-go/pull/1279/files#diff-098f827fb283aa9a2a88906bc2e691e6R43).

Reply:
Yup move to load.go as you suggested (i.e. only initialize it if we've determined we are using kubernetes datastore). Then I think we are good to go. Thanks!